src/core/contract_address/casm_contract_address.rs

```diff
@@ -0,0 +1,190 @@
+        let contract_class: CasmContractClass = serde_json::from_reader(reader).unwrap();
+
+        assert_eq!(
+            compute_casm_class_hash(&contract_class).unwrap(),
+            felt_str!(
+                "3a4f00bf75ba3b9230a94f104c7a4605a1901c4bd475beb59eeeeb7aceb9795",
+                16
+            )
+        );
+    }
+
+    #[test]
+    fn test_compute_casm_class_hash_deploy() {
+        // Open the file in read-only mode with buffer.
+        let file = File::open("starknet_programs/cairo1/deploy.casm").unwrap();
+        let reader = BufReader::new(file);
+
+        // Read the JSON contents of the file as an instance of `User`.
+        let contract_class: CasmContractClass = serde_json::from_reader(reader).unwrap();
+
+        assert_eq!(
+            compute_casm_class_hash(&contract_class).unwrap(),
+            felt_str!(
+                "3bd56f1c3c1c595ac2ee6d07bdedc027d09df56235e20374649f0b3535c1f15",
+                16
+            )
+        );
+    }
+
+    #[test]
+    fn test_compute_casm_class_hash_fibonacci() {
+        // Open the file in read-only mode with buffer.
+        let file = File::open("starknet_programs/cairo1/fibonacci.casm").unwrap();
+        let reader = BufReader::new(file);
+
+        // Read the JSON contents of the file as an instance of `User`.
+        let contract_class: CasmContractClass = serde_json::from_reader(reader).unwrap();
+
+        assert_eq!(
+            compute_casm_class_hash(&contract_class).unwrap(),
+            felt_str!(
+                "44f12e6e59232e9909d7428b913b3cc8d9059458e5027740a3ccdbdc4b1ffd2",
+                16
+            )
+        );
+    }
+
+    #[test]
+    fn test_compute_casm_class_hash_factorial() {
+        // Open the file in read-only mode with buffer.
+        let file = File::open("starknet_programs/cairo1/factorial.casm").unwrap();
+        let reader = BufReader::new(file);
+
+        // Read the JSON contents of the file as an instance of `User`.
+        let contract_class: CasmContractClass = serde_json::from_reader(reader).unwrap();
+
+        assert_eq!(
+            compute_casm_class_hash(&contract_class).unwrap(),
+            felt_str!(
+                "189a9b8b852aedbb225aa28dce9cfc3133145dd623e2d2ca5e962b7d4e61e15",
+                16
+            )
+        );
+    }
+
+    #[test]
+    fn test_compute_casm_class_hash_emit_event() {
+        // Open the file in read-only mode with buffer.
+        let file = File::open("starknet_programs/cairo1/emit_event.casm").unwrap();
+        let reader = BufReader::new(file);
+
+        // Read the JSON contents of the file as an instance of `User`.
+        let contract_class: CasmContractClass = serde_json::from_reader(reader).unwrap();
+
+        assert_eq!(
+            compute_casm_class_hash(&contract_class).unwrap(),
+            felt_str!(
+                "3335fe731ceda1116eda8bbc2e282953ce54618309ad474189e627c59328fff",
+                16
+            )
+        );
+    }
```

Comment:
Should we also add a test with a Cairo 2 contract?

Reply:
Makes sense.

Reply:
Done.